### DISCUSSION TITLE
Bug report: Why are some buttons not working?

### DISCUSSION THREAD

**Discussion by SuperVlogger94** [MinicraftPlus/minicraft-plus-revived#651] [Q&A] [answered]
**Describe the bug**
Movement on W A S D, opening inventory, etc. do not work. where symbols are used to activate

**Desktop**
 - OS Windows 10 pro, OS Version 21H2
 - Version 2.1.3

**Additional context**
Also I can not return the control to the standard, but the keyboard is in working condition

**Comment by TheBigEye** [comment]
Try pressing `Windows` + `R` keys, that will open a window, there type `%APPDATA%/playminicraft/mods/Minicraft_Plus/`, press `Enter`

The file explorer will open, there delete the `Preferences.json` file (this will delete the game configuration, making that when it is started again, it will generate the normal configuration again)

**Comment by SuperVlogger94** [comment]
The controls are back to normal, but the problem with opening inventory, moving to WASD, etc. not resolved

**Comment by Litorom** [comment]
You are pressing the default key “X” right?

**Comment by BenCheung0422** [comment]
Is the key binding shown in options menu correct?

**Comment by SuperVlogger94** [comment]
Yes, specified correctly

**Comment by BenCheung0422** [comment]
Can you describe your actions? Like pressing what buttons what is happened, etc.

**Comment by SuperVlogger94** [comment]
After installing the update, WASD, opening inventory, etc. stopped working for me, I already reinstalled the game

**Comment by BenCheung0422** [comment]
Do you mean that there is no reaction when pressing the keyboard?

**Comment by SuperVlogger94** [comment]
Yes, but only for symbols

**Comment by BenCheung0422** [comment, marked as answer]
This is mostly possibly because of the IME.
Troubleshooting:
1. Make sure the IME is on English mode.
2. Turn input method to English IME if it is not.
3. Try using previos version and test if there is still problem.
4. Try using another version of JRE/JDK or reinstalling JRE/JDK.

**Comment by SuperVlogger94** [comment]
I changed the IME to English and it worked, thank you very much, but the fact that the reaction only works on the English IME is at least strange

**Comment by BenCheung0422** [comment]
Glad to hear. I think the reason is the IME input uses different control charactors, and the game only detects A-Z key input as the keys.

**Comment by swyat1911** [comment]
i have the same problem
